### PR TITLE
add file validation logic

### PR DESCRIPTION
**Pull request by mfdeveloper508**

### CODE REVIEW

File: file/file_instance.go [hunk 1]
```diff
@@ -72,11 +78,87 @@ func (f *fileInstance) GetDataRecords() []lib.Record {
...
+		information, err = f.generatorTrailer()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	fromFields := reflect.ValueOf(information).Elem()
+	toFields := reflect.ValueOf(trailer).Elem()
+	for i := 0; i < fromFields.NumField(); i++ {
+		fieldName := fromFields.Type().Field(i).Name
```
Comment (adamdecaf) [inline file/file_instance.go:101]: Are there `nil` checks to perform here so we avoid a panic?

File: file/file_instance.go [hunk 2]
```diff
@@ -72,11 +78,87 @@ func (f *fileInstance) GetDataRecords() []lib.Record {
...
+		fromField := fromFields.FieldByName(fieldName)
+		toField := toFields.FieldByName(fieldName)
+		if fromField.IsValid() && toField.CanSet() {
+			toField.Set(fromField)
+		}
+	}
+
+	if f.format == PackedFileFormat {
+		segment := trailer.(*lib.PackedTrailerRecord)
```
Comment (adamdecaf) [inline file/file_instance.go:110]: Can we check the type cast here? `segment, ok := trailer.(*lib.PackedTrailerRecord)`

Comment (mfdeveloper508) [reply file/file_instance.go]: fixed

File: file/file_instance.go (hunk 1 above)
Comment (mfdeveloper508) [reply file/file_instance.go:101]: fromFields.Type().Field(i) will return struct, not pointer

**Pull request merged**